### PR TITLE
Added readme for setting guidelines for plugin development 

### PR DESCRIPTION
**Pull request by VincentvanMedenbach**

### CODE REVIEW

**Review by Daan0709 — APPROVED** [review]

File: src/apisix/plugins/readme.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+For developing custom plugins, follow these guidelines.
+
+## 1. Create a test environment
+
+Create a single dedicated test environment that can be used for:
+
+- Automated testing pipelines
+- Local development
+
+To do this, create a folder under `tests` with the plugin name. Include a `Dockerfile` and the relevant configuration files.
```
Comment (vdegans) [inline src/apisix/plugins/readme.md]:
```suggestion
To do this, create a folder under `tests` using the plugin name. Include a `docker-compose.yaml` and the relevant configuration files.
```

File: src/apisix/plugins/readme.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+}
+```
+
+## 4. Add requests and tests
+
+Add relevant requests and tests to the Bruno collection.
+
+You do not need to cover every edge case, but include at least:
+
+- A happy path
```
Comment (vdegans) [inline src/apisix/plugins/readme.md]:
```suggestion
- A happy scenario
```

File: src/apisix/plugins/readme.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+- A failure path
```
Comment (vdegans) [inline src/apisix/plugins/readme.md]:
```suggestion
- A failure scenario
```

File: src/apisix/plugins/readme.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+For an auth plugin, for example:
```
Comment (vdegans) [inline src/apisix/plugins/readme.md]:
```suggestion
For example, for an auth plugin:
```

**Review by vdegans — CHANGES_REQUESTED** [review]

**Review by vdegans — APPROVED** [review]
